File: android/app/src/main/java/app/organicmaps/sdk/Router.java
package app.organicmaps.sdk;

import androidx.annotation.NonNull;

public enum Router
{
  Vehicle(0),
  Pedestrian(1),
  Bicycle(2),
  Transit(3),
  Ruler(4);

  Router(int type)
  {
    this.type = type;
  }

  public static void set(@NonNull Router routerType)
  {
    nativeSet(routerType.type);
  }

  public static Router get()
  {
    return valueOf(nativeGet());
  }

  public static Router getLastUsed()
  {
    return valueOf(nativeGetLastUsed());
  }

  public static Router getBest(double srcLat, double srcLon, double dstLat, double dstLon)
  {
    return Router.values()[nativeGetBest(srcLat, srcLon, dstLat, dstLon)];
  }

  public static Router valueOf(int type)
  {
    return Router.values()[type];
  }

  private final int type;

  private static native void nativeSet(int routerType);

  private static native int nativeGet();

  private static native int nativeGetLastUsed();

  private static native int nativeGetBest(double srcLat, double srcLon,
                                          double dstLat, double dstLon);
}

File: libs/routing/router.cpp
#include "routing/router.hpp"

namespace routing
{
std::string ToString(RouterType type)
{
  switch (type)
  {
  case RouterType::Vehicle: return "vehicle";
  case RouterType::Pedestrian: return "pedestrian";
  case RouterType::Bicycle: return "bicycle";
  case RouterType::Transit: return "transit";
  case RouterType::Ruler: return "ruler";
  case RouterType::Count: return "count";
  }
  ASSERT(false, ());
  return "Error";
}

RouterType FromString(std::string const & str)
{
  if (str == "vehicle")
    return RouterType::Vehicle;
  if (str == "pedestrian")
    return RouterType::Pedestrian;
  if (str == "bicycle")
    return RouterType::Bicycle;
  if (str == "transit")
    return RouterType::Transit;
  if (str == "ruler")
    return RouterType::Ruler;

  ASSERT(false, ("Incorrect routing string:", str));
  return RouterType::Vehicle;
}

std::string DebugPrint(RouterType type)
{
  return ToString(type);
}
}  //  namespace routing
